config/main.py

```diff
@@ -1804,13 +1815,43 @@ def add_snmp_agent_address(ctx, agentip, port, vrf):
     """Add the SNMP agent listening IP:Port%Vrf configuration"""
 
     #Construct SNMP_AGENT_ADDRESS_CONFIG table key in the format ip|<port>|<vrf>
+    if not clicommon.is_ipaddress(agentip):
+        click.echo("Invalid IP address")
+        return False
+    config_db = ctx.obj['db']
+    if not vrf:
+        entry = config_db.get_entry('MGMT_VRF_CONFIG', "vrf_global")
+        if entry and entry['mgmtVrfEnabled'] == 'true' :
+            click.echo("ManagementVRF is Enabled. Provide vrf.")
+            return
+    found = 0
+    ip = ipaddress.ip_address(agentip)
+    for intf in netifaces.interfaces():
+        ipaddresses = netifaces.ifaddresses(intf)
+        if ip_family[ip.version] in ipaddresses:
+            for ipaddr in ipaddresses[ip_family[ip.version]]:
+                if agentip == ipaddr['addr']:
+                    found = 1
+                    break;
+        if found == 1:
+            break;
+    if found == 0:
```

Comment:
>if found == 0: [](start = 4, length = 14)

you can use `for-else` syntax to reduce duplicated test.

Reply:
Resolved, submitted patch